xds/internal/xdsclient/tlscreds/bundle_ext_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package tlscreds_test
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/internal/stubserver"
+	"google.golang.org/grpc/internal/testutils/xds/e2e"
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/testdata"
+	"google.golang.org/grpc/xds/internal/xdsclient/tlscreds"
+)
+
+func TestValidTlsBuilder(t *testing.T) {
+	tests := []struct {
+		name string
+		jd   string
+	}{
+		{"Absent configuration", `null`},
+		{"Empty configuration", `{}`},
+		{"Only CA certificate chain", `{"ca_certificate_file": "foo"}`},
+		{"Only private key and certificate chain", `{"certificate_file":"bar","private_key_file":"baz"}`},
+		{"CA chain, private key and certificate chain", `{"ca_certificate_file":"foo","certificate_file":"bar","private_key_file":"baz"}`},
+		{"Only refresh interval", `{"refresh_interval": "1s"}`},
+		{"Refresh interval and CA certificate chain", `{"refresh_interval": "1s","ca_certificate_file": "foo"}`},
+		{"Refresh interval, private key and certificate chain", `{"refresh_interval": "1s","certificate_file":"bar","private_key_file":"baz"}`},
+		{"Refresh interval, CA chain, private key and certificate chain", `{"refresh_interval": "1s","ca_certificate_file":"foo","certificate_file":"bar","private_key_file":"baz"}`},
+		{"Unknown field", `{"unknown_field": "foo"}`},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			msg := json.RawMessage(test.jd)
+			if _, err := tlscreds.NewBundle(msg); err != nil {
+				t.Errorf("NewBundle(%s): expected no error but got: %s", test.jd, err)
```

Comment:
Please specify got before want. See https://google.github.io/styleguide/go/decisions#got-before-want

So, this would be something like:

`t.Errorf("NewBundle(%s) returned error %s when expected to succeed", test.jd, err)`

Reply:
Done.